x/interchainqueries/keeper/msg_server.go

```diff
@@ -82,12 +89,15 @@ func (k msgServer) RemoveInterchainQuery(goCtx context.Context, msg *types.MsgRe
 		return nil, sdkerrors.Wrapf(err, "failed to get query by query id: %v", err)
 	}
 
-	if query.GetOwner() != msg.GetSender() {
+	timoutBlock := query.LastSubmittedResultLocalHeight + k.GetParams(ctx).QuerySubmitTimeout
+	if uint64(ctx.BlockHeight()) <= timoutBlock && query.GetOwner() != msg.GetSender() {
 		ctx.Logger().Debug("RemoveInterchainQuery: authorization failed",
 			"msg", msg)
 		return nil, sdkerrors.Wrap(sdkerrors.ErrUnauthorized, "authorization failed")
 	}
+
 	k.RemoveQueryByID(ctx, query.Id)
+	k.MustPayOutDeposit(ctx, query.GetDepositCoin(), msg.GetSigners()[0])
```

Comment:
unhandled error

Reply:
yes, because it should `panic`. Anyway let's think about it, how it is better to cover this situation.

Reply:
I think we can just do not return anything from `MustPayOutDeposit`, since there is only one function call which is covered by `panic` anyway